validator/resource_value_pattern_test.rego

```diff
@@ -0,0 +1,142 @@
+package templates.gcp.GCPResourceValuePatternConstraintV1
+
+all_required_violations[violation] {
+	resource := data.test.fixtures.assets.projects[_]
+	constraint := data.test.fixtures.constraints.resource_value_pattern_required
+
+	issues := deny with input.asset as resource
+		 with input.constraint as constraint
+
+	violation := issues[_]
+}
+
+all_not_required_violations[violation] {
+	resource := data.test.fixtures.assets.projects[_]
+	constraint := data.test.fixtures.constraints.resource_value_pattern_not_required
+
+	issues := deny with input.asset as resource
+		 with input.constraint as constraint
+
+	violation := issues[_]
+}
+
+all_no_pattern_violations[violation] {
+	resource := data.test.fixtures.assets.projects[_]
+	constraint := data.test.fixtures.constraints.resource_value_pattern_no_pattern
+
+	issues := deny with input.asset as resource
+		 with input.constraint as constraint
+
+	violation := issues[_]
+}
+
+# Confirm total violations count
+test_resource_value_pattern_required_violations_count {
+	count(all_required_violations) == 2
```

Comment:
instead of relying on count, can we verify the resource field in violation details to make sure that we're getting the expected violations?

Reply:
I updated the tests and added a couple more as well as asserting the resource names.